man/man7/getty.7
.pa 1
.he '12/11/72''GETTY (VII)'
.tr |
.ti 0
.nf
NAME		getty  -- set typewriter mode and get user's name
.fi
.sp
.ti 0
SYNOPSIS	/etc/getty
.sp
.ti 0
DESCRIPTION
.br
.in 8
getty_____
is invoked
by init (VII)
immediately after a typewriter is opened
following a dial-in.
The user's login name is read and the login(I) command
is called with this name as an argument.
While reading this name getty_____
attempts to adapt the system to the speed and type of terminal
being used.
.sp
getty_____ initially sets the speed of the interface to 150 baud,
specifies that raw mode is to be used (break on every character),
that echo is to be suppressed, and either parity
allowed.
It types the "login:" message
(which includes the characters which put the 37 Teletype
terminal into full-duplex and unlock its keyboard).
Then the user's name is read, a character at a time.
If a null character is received, it is assumed to be the result
of the user pushing the "break" ("interrupt") key.
The speed is then
changed to 300 baud and the "login:" is typed again,
this time with the appropriate
sequence which puts a GE TermiNet 300 into full-duplex.
This sequence is acceptable to other 300 baud terminals also.
If a subsequent null character is received,
the speed is changed again.
The general approach is to cycle through a set of speeds in
response to null characters caused by breaks.
The sequence at this installation is 150, 300, and 134.5 baud.
.sp
Detection of IBM|2741s is accomplished while the speed is
set to 150 baud.
The user sends a 2741 style "eot"
character by pushing the attention key or by typing return;
at 150 baud, this character looks like the ascii "~" (174988).
Upon receipt of the "eot", the system is set to operate 2741s
and a "login: " message is typed.

The user's name is terminated by a new-line or
carriage-return character.
The latter results in the system being set to
to treat carriage returns appropriately (see stty(II)).

The user's name is scanned to see if
it contains any lower-case alphabetic characters; if not,
and if the name is nonempty, the
system is told to map any future upper-case characters
into the corresponding lower-case characters.
Thus UNIX is usable from upper-case-only terminals.

Finally, login is called with the user's name as argument.
.sp
.in 16
.ti 0
FILES		--
.sp
.ti 0
SEE ALSO	init(VII), login(I), stty(II)


named/CHANGES
This is version 4.8.1 of bind.  It is a test release that includes
version 4.8 with fixes, asynchronous zone transfer and better reload
capabilities.  Although it is currently described as a test release,
it is believed to be reasonably stable and more usable than the currently-
released version, 4.8.  The changes of note are:

  o The asynchronous zone transfer code previously posted to the bind
    mailing list has been integrated, completed and tested.  There are
    a number of changes from the version posted, including fixes to
    allow top-level domains to work and a simplification of the timer
    code.

  o The code for reloading the server has been changed so that only
    primary zones master files that have changed are reloaded.  The
    cache and secondary zones are not flushed, and the sortlist, domain,
    etc. are reset to correspond to the boot file contents.

  o Several bugs have been fixed: the name "*" is not interpreted as
    a wildcard in cached zones, only in primary zones.  Secondary servers
    no longer decrement the time-to-live of records by the time since
    they verified the zone with the master; as a result, they never
    hand out nameserver referrals with too short a TTL to be usable.
    A bug was fixed that caused secondary servers with out-of-date
    zones to return empty answers between the actual expiration time
    and the next timeout.

There are several other bugs that have been reported but have not yet
been fixed.  In addition, the next regular release of named will
support negative caching, but this has not been integrated.

I would appreciate receiving feedback on this release; in particular,
problems (or lack of problems) when installing on various systems.
I attempted to update the SysV code when integrating the zone-transfer,
but haven't tested it.

The notes from version 4.8 follow.

----------
Welcome to version 4.8 of bind.

There have been several changes to the named boot file (/etc/named.boot)
of which you should be aware.  The "domain" line for each zone is no longer
needed, but one such line may still be used to specify a default domain
to be used for queries containing names with only a single component.
The term "suffixes", which was added in version 4.7alpha, has been removed.

The manual page on named (named.8) has been updated to reflect all
these changes.  Please read this and look at the example files
before installation.  You should also note the changes in the
resolver code to support non-fully-qualified addresses and per-user
host aliases.  See hostname(7) for an overview.  Two new routines
have been added to the resolver library since the last test release:
res_query formulates a query, sends it, waits for a response and does
preliminary error checking; res_search implements the search rules
of gethostbyname using res_query.

The MX lookup routine in sendmail has been modified to use res_search.  
Also, dn_skip takes an additional parameter and has been renamed
to dn_skipname.  While old sendmail binaries will work with the new
version of bind, because of these changes, it is desirable to install
new sendmail sources and recompile sendmail.  Do not rebuild sendmail 
from old sources.   The new sendmail is on ucbarpa.Berkeley.EDU for
anonymous FTP from pub/4.3/sendmail.MX.tar and pub/4.3/sendmail.MX.tar.Z.

There have been numerous changes to named, fixing most of the known
bugs that can be fixed without major structural changes in the server.
Several server configurations that failed before should now work.
Certain robustness problems have been fixed, in particular bounds-
checking when processing incoming packets.  Two changes have been made
in preparation for negative caching: SOA records are sent in the authority
section in negative responses with NXDOMAIN set, and a bug was fixed that
caused confusion and repeated requests if a response had no error, no answer
and an SOA in the authority section.  As such responses are already sent
by other servers, and will be sent by the next release of BIND, it is
important that all sites upgrade to this version as quickly as possible.

The root "hint" cache and cache file remain the largest problem area,
along with named's naivete in accepting bogus server's data.
These will be addressed in the next release, along with asynchronous
zone transfers, intelligent reloading of zone files, faster startup,
and caching of negative responses.

This version (4.8) will replace the last officially released version (4.5).
Version 4.5 has a serious bug that causes the generation of a continuous
stream of bogons to the root domain servers (bogus queries with the query
response bit set and possibly garbage for nsid and rcode).  It is imperative
that these versions of named be replaced as fast as possible.  We urge you to
field 4.8 quickly, for the sake of the root domain servers.

	Mike Karels
	Jean Wood
	bind@ucbarpa.Berkeley.EDU


cmd/getty.s
/ getty --  get name and tty mode
/ for initialization

/ cycle through speeds and "login:" messages
/ summarized in itab

stty = 31.

	sys	quit; 0
	sys	intr; 0
0:
	jsr	r5,nextspeed
1:
	mov	$name,r5
2:
	jsr	r5,getc
	cmp	r0,$174
	beq	5f
	cmp	r0,$176
	beq	5f
	cmp	r0,$'\n
	beq	1f
	cmp	r0,$'\r
	beq	4f
	cmp	r0,$'@
	beq	1b
	cmp	r0,$'#
	bne	3f
	cmp	r5,$name
	blos	2b
	dec	r5
	br	2b
3:
	movb	r0,(r5)+
	br	2b
4:
	bis	$20,flags		/cr bit
	mov	$1,r0
	sys	write; nl; 1
	br	2f
5:
	mov	$tab2741,itabp
	inc	nowr
	br	0b
1:
	mov	$1,r0
	sys	write; cr; 1
2:
	clrb	(r5)+

/ determine whether terminal is upper-case only

	cmp	r5,$name+1
	bhi	1f
	bic	$4,flags	/no data-assume lc
1:
	mov	$name,r5
1:
	movb	(r5)+,r0
	beq	1f
	cmp	r0,$'A
	blo	2f
	cmp	r0,$'Z
	bhi	2f
	add	$40,r0		/ map to lc
	movb	r0,-1(r5)
	br	1b
2:
	cmp	r0,$'a
	blo	1b
	cmp	r0,$'z
	bhi	1b
	bic	$4,flags
	br	1b
1:
	clr	r0
	mov	fstate,r4
	bis	flags,4(r4)
	sys	stty; fstate: ..

go:
	sys	exec; login; loginp
	sys	exit

getc:
	clr	r0
	sys	read; ch; 1
	tst	r0
	beq	done
	mov	ch,r2
	beq	1f
getc1:
	cmp	r2,$174
	bhis	3f
	tst	nowr
	bne	3f
	mov	$1,r0
	sys	write; ch; 1
3:
	mov	r2,r0
	rts	r5
1:
	dec	$0		/ wait a while
	bne	1b
	mov	$name,(sp)
	jsr	r5,nextspeed
2:
	clr	r0		/ flush nulls
	sys	read; ch; 1
	tst	r0
	beq	done
	movb	ch,r2
	beq	2b
	br	getc1

done:
	sys	exit

nextspeed:
	mov	itabp,r1
	mov	(r1)+,0f
	bne	1f
	mov	$itab,itabp
	br	nextspeed
1:
	clr	r0
	sys	stty; 0:..
	bes	go
	mov	(r1)+,-(sp)
	mov	(r1)+,fstate
	mov	r1,itabp
	mov	(sp)+,r1
1:
	movb	(r1)+,ch
	beq	1f
	mov	$1,r0
	sys	write; ch; 1
	br	1b
1:
	rts	r5

itabp:	itab
loginp:	login
	name
	0

itab:
	itty37; ttymes; tty37
	itn300; tnmes;  tn300
tab2741: i2741; m2741; f2741
	0

itty37:	511; 511; 340	/ any parity, raw, 150 baud
tty37:	511; 511; 210	/ 37 parity, echo, 150 baud
itn300:	521; 521; 340	/ any parity, raw, cr, 300 baud
tn300:	521; 521; 310	/ any parity, echo, 300 baud
i2741:	1501; 501; 100540	/134 bits, 2741, raw, first time
f2741:	1501; 501; 500	/134 bps, 2741

	0
m2741:	<\nlogin: \0>

ttymes:
	<\n\r\p:\alogin: \0>
tnmes:
	<\n\r\p;login: \0>

login:	</bin/login\0>
	.even

nl:	<\n>
cr:	<\r>

flags:	004	/ upper case map

	.bss
ch:	.=.+2
nowr:	.=.+2
name:	.=.+32.
f2741
	0

itty37:	511; 511; 340	/ any parity, raw, 150 baud
tty37:	511; 511; 210	/ 37 parity, echo, 150 baud
itn300:	521; 521; 340	/ any parity, raw, cr, 300 baud
tn300:	521; 521; 310	/ any parity, echo, 300 baud
i2741:	1501; 501; 100540	/134 bits, 2741, raw, first time
f2741:	1501; 501; 500	/134 bps, 2741

	0
m2741:	<\nlogin: \0>

ttymes:
	<\n\r\p:\alogin: \0>
tnmes:
	<\n\r\p;log